arches/app/src/components/ControlledListManager/ItemImages.vue

```diff
@@ -0,0 +1,130 @@
+<script setup lang="ts">
+import arches from "arches";
+import Cookies from "js-cookie";
+import { inject } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import FileUpload from "primevue/fileupload";
+import { useToast } from "primevue/usetoast";
+
+import ImageEditor from "@/components/ControlledListManager/ImageEditor.vue";
+
+import { ARCHES_CHROME_BLUE } from "@/theme.ts";
+import { itemKey, ERROR } from "@/components/ControlledListManager/constants.ts";
+
+import type { Ref } from "vue";
+import type { ControlledListItem } from "@/types/ControlledListManager";
+import type {
+    FileUploadBeforeSendEvent,
+    FileUploadErrorEvent,
+    FileUploadUploadEvent,
+} from "primevue/fileupload";
+
+const item = inject(itemKey) as Ref<ControlledListItem>;
+const completed = 'Completed';
+
+const { $gettext } = useGettext();
+const toast = useToast();
+
+const addHeader = (event: FileUploadBeforeSendEvent) => {
+    const token = Cookies.get("csrftoken");
+    if (token) {
+        event.xhr.setRequestHeader("X-CSRFToken", token);
```

Comment:
I'm doing it this way because I'm adhering to the interface for PrimeVue's FileUpload component (see where it takes props for file types, upload url, file limit etc). I think it's okay to deviate from the other api.ts pattern when we are using the FileUpload component. I've liked working with it, btw.